### PR TITLE
Fix #1161, #1130 Apply and check standard code format 

### LINKED ISSUE #1130: Apply style formatting (release candidate prep)
**Is your feature request related to a problem? Please describe.**
Code has diverged from automated style.

**Describe the solution you'd like**
Apply automated style formatting.

**Describe alternatives you've considered**
None

**Additional context**
Suggest adding as a check in workflow until release.  Style has been very stable, should be manageable to enforce at least in the short term (release candidate and release prep).  Could enforce just on RC branch(es) if there's an issue w/ general enforcement.

**Requester Info**
Jacob Hageman - NASA/GSFC

### LINKED ISSUE #1161: Add format check to workflow
**Is your feature request related to a problem? Please describe.**
No indication of software format non-compliance on pull requests

**Describe the solution you'd like**
Add format check to workflow

**Describe alternatives you've considered**
None

**Additional context**
See #1130 for applying the standard formatting

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by astrogeco**
**Describe the contribution**
Fix #1161 
Fix #1130 

**Testing performed**
See checks tab <https://github.com/nasa/cFE/pull/1219/checks>
Bundle (fork) test: <https://github.com/astrogeco/cFS/pull/8/checks>

**Expected behavior changes**
Adds new github actions workflow that checks format. This workflow runs on both "push" and "pull_request" for all branches so at the moment it will trigger twice for any branch that is part of a pull request

**System(s) tested on**
Ubuntu 18.04

**Additional context**
Will create a tag after merge

**Third party code**
None

### CODE REVIEW

File: modules/core_api/fsw/inc/cfe_es.h [hunk 1]
```diff
@@ -219,8 +217,9 @@ void CFE_ES_Main(uint32 StartType, uint32 StartSubtype, uint32 ModeId , const ch
 **          None
 **
 ** \param[in]  ResetType    Identifies the type of reset desired.  Allowable settings are:
-**                          \arg #CFE_PSP_RST_TYPE_POWERON     - Causes all memory to be cleared 
-**                          \arg #CFE_PSP_RST_TYPE_PROCESSOR   - Attempts to retain volatile disk, critical data store and user reserved memory. 
+**                          \arg #CFE_PSP_RST_TYPE_POWERON     - Causes all memory to be cleared
+**                          \arg #CFE_PSP_RST_TYPE_PROCESSOR   - Attempts to retain volatile disk, critical data store
+*and user reserved memory.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:222]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 2]
```diff
@@ -494,8 +492,10 @@ void  CFE_ES_IncrementTaskCounter(void);
 **          None
 **
 ** \param[in, out]   ResetSubtypePtr    Pointer to \c uint32 type variable in which the Reset Sub-Type will be stored.
-**                                 The caller can set this pointer to NULL if the Sub-Type is of no interest. \n *ResetSubtypePtr If the provided pointer was not \c NULL, the Reset Sub-Type is stored at the given address.
-**                                 For a list of possible Sub-Type values, see \link #CFE_PSP_RST_SUBTYPE_POWER_CYCLE "Reset Sub-Types" \endlink.
+**                                 The caller can set this pointer to NULL if the Sub-Type is of no interest. \n
+**ResetSubtypePtr If the provided pointer was not \c NULL, the Reset Sub-Type is stored at the given address.
+**                                 For a list of possible Sub-Type values, see \link #CFE_PSP_RST_SUBTYPE_POWER_CYCLE
+*"Reset Sub-Types" \endlink.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:498]: Fix line splits

File: modules/core_api/fsw/inc/cfe_es.h [hunk 3]
```diff
@@ -697,14 +697,15 @@ CFE_Status_t CFE_ES_GetAppInfo(CFE_ES_AppInfo_t *AppInfo, CFE_ES_AppId_t AppId);
...
 **
 ** \par Assumptions, External Events, and Notes:
 **        None
 **
-** \param[out]   TaskInfo     Pointer to a \c CFE_ES_TaskInfo_t structure that holds the specific 
-**                            task information. *TaskInfo is the filled out \c CFE_ES_TaskInfo_t structure containing the 
+** \param[out]   TaskInfo     Pointer to a \c CFE_ES_TaskInfo_t structure that holds the specific
+**                            task information. *TaskInfo is the filled out \c CFE_ES_TaskInfo_t structure containing
+*the
 **                            Task Name, Parent App Name, Parent App ID among other fields.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:709]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 4]
```diff
@@ -811,36 +812,37 @@ int32 CFE_ES_GetModuleInfo(CFE_ES_AppInfo_t *ModuleInfo, CFE_ResourceId_t Resour
...
 ** \par Description
-**        This routine creates a new task (a separate execution thread) owned by the calling Application.  
+**        This routine creates a new task (a separate execution thread) owned by the calling Application.
 **
 ** \par Assumptions, External Events, and Notes:
 **        None
 **
-** \param[in, out]   TaskIdPtr     A pointer to a variable that will be filled in with the new task's ID. *TaskIdPtr is the Task ID of the newly created child task.
+** \param[in, out]   TaskIdPtr     A pointer to a variable that will be filled in with the new task's ID. *TaskIdPtr is
+*the Task ID of the newly created child task.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:828]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 5]
```diff
@@ -926,7 +924,8 @@ CFE_Status_t CFE_ES_GetTaskName(char *TaskName, CFE_ES_TaskId_t TaskId, size_t B
 ** \par Assumptions, External Events, and Notes:
 **        None
 **
-** \param[in]   TaskId     The task ID previously obtained when the Child Task was created with the #CFE_ES_CreateChildTask API.
+** \param[in]   TaskId     The task ID previously obtained when the Child Task was created with the
+*#CFE_ES_CreateChildTask API.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:928]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 6]
```diff
@@ -1069,17 +1067,18 @@ void CFE_ES_ProcessAsyncEvent(void);
...
 **        This routine allocates a block of memory in the Critical Data Store and associates it with
-**        the calling Application.  The memory can survive an Application restart as well as a Processor Reset. 
+**        the calling Application.  The memory can survive an Application restart as well as a Processor Reset.
 **
 ** \par Assumptions, External Events, and Notes:
 **        None
 **
-** \param[in, out]   HandlePtr   Pointer Application's variable that will contain the CDS Memory Block Handle. *HandlePtr is the handle of the CDS block that can be used in 
+** \param[in, out]   HandlePtr   Pointer Application's variable that will contain the CDS Memory Block Handle.
+**HandlePtr is the handle of the CDS block that can be used in
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:1076]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 7]
```diff
@@ -1194,7 +1192,8 @@ CFE_Status_t CFE_ES_CopyToCDS(CFE_ES_CDSHandle_t Handle, void *DataToCopy);
 **
 ** \param[in]   Handle             The handle of the CDS block that was previously obtained from #CFE_ES_RegisterCDS.
 **
-** \param[in, out]   RestoreToMemory    A Pointer to the block of memory that is to be restored with the contents of the CDS. *RestoreToMemory is the contents of the specified CDS.
+** \param[in, out]   RestoreToMemory    A Pointer to the block of memory that is to be restored with the contents of the
+*CDS. *RestoreToMemory is the contents of the specified CDS.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:1196]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 8]
```diff
@@ -1217,14 +1216,15 @@ CFE_Status_t CFE_ES_RestoreFromCDS(void *RestoreToMemory, CFE_ES_CDSHandle_t Han
...
+**        created by this routine is processed, no mutex handling is performed.
 **
 ** \par Assumptions, External Events, and Notes:
 **        -# The size of the pool must be an integral number of 32-bit words
 **        -# The start address of the pool must be 32-bit aligned
 **        -# 168 bytes are used for internal bookkeeping, therefore, they will not be available for allocation.
 **
-** \param[in, out]   PoolID   A pointer to the variable the caller wishes to have the memory pool handle kept in. *PoolID is the memory pool handle.
+** \param[in, out]   PoolID   A pointer to the variable the caller wishes to have the memory pool handle kept in.
+**PoolID is the memory pool handle.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:1227]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 9]
```diff
@@ -1274,28 +1275,32 @@ CFE_Status_t CFE_ES_PoolCreate(CFE_ES_MemHandle_t *PoolID, uint8 *MemPtr, size_t
...
+**        This routine initializes a pool of memory supplied by the calling application.
 **
 ** \par Assumptions, External Events, and Notes:
 **        -# The size of the pool must be an integral number of 32-bit words
 **        -# The start address of the pool must be 32-bit aligned
 **        -# 168 bytes are used for internal bookkeeping, therefore, they will not be available for allocation.
 **
-** \param[in, out]   PoolID      A pointer to the variable the caller wishes to have the memory pool handle kept in. *PoolID is the memory pool handle.
+** \param[in, out]   PoolID      A pointer to the variable the caller wishes to have the memory pool handle kept in.
+**PoolID is the memory pool handle.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:1286]: Fix line split

File: modules/core_api/fsw/inc/cfe_es.h [hunk 10]
```diff
@@ -1274,28 +1275,32 @@ CFE_Status_t CFE_ES_PoolCreate(CFE_ES_MemHandle_t *PoolID, uint8 *MemPtr, size_t
...
 **
 ** \param[in]   BlockSizes     Pointer to an array of sizes to be used instead of the default block sizes specified by
-**                             #CFE_PLATFORM_ES_MEM_BLOCK_SIZE_01 through #CFE_PLATFORM_ES_MAX_BLOCK_SIZE.  If the pointer is equal to NULL,
+**                             #CFE_PLATFORM_ES_MEM_BLOCK_SIZE_01 through #CFE_PLATFORM_ES_MAX_BLOCK_SIZE.  If the
+*pointer is equal to NULL,
 **                             the default block sizes are used.
 **
-** \param[in]   UseMutex       Flag indicating whether the new memory pool will be processing with mutex handling or not.
+** \param[in]   UseMutex       Flag indicating whether the new memory pool will be processing with mutex handling or
+*not.
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_es.h:1303]: Fix line splits

**Review by skliper — COMMENTED** [review]
Many splits should be cleaned up... Stopped after identifying 10

**Comment by skliper** [comment]
Note - could just merge as-is and clean up in follow-on.  This is all due to the original having violated the 120 character rule, so it was bad then and it's bad now.  The critical point is to get the bulk format in since in general it's a huge consistency improvement.

**Comment by astrogeco** [comment]
> Note - could just merge as-is and clean up in follow-on. This is all due to the original having violated the 120 character rule, so it was bad then and it's bad now. The critical point is to get the bulk format in since in general it's a huge consistency improvement.

THe problem is mainly ensuring the two `**` and a space after them right? So we'd like this 
```
** \param[in, out]   PoolID      A pointer to the variable the caller wishes to have the memory pool handle kept in.
**PoolID is the memory pool handle.
```

to look like this (at the very least)

```
** \param[in, out]   PoolID      A pointer to the variable the caller wishes to have the memory pool handle kept in.
** PoolID is the memory pool handle.
```

or in a perfect world look like

```
** \param[in, out]   PoolID      A pointer to the variable the caller wishes to have the memory pool handle kept in.
**                                             PoolID is the memory pool handle.
```

**Comment by jphickey** [comment]
This is where I recommend (and have recommended in the past) going to a single "*" char rather than "**" on the block comments.  I think most tools/auto-formatters work better with this pattern, the double * is a little unusual....  (Might be possible to fix this with a sed script)

**Comment by astrogeco** [comment]
What's the history and reasoning behind the `**` style?

**Comment by jphickey** [comment]
No reason AFAIK, someone used it at one time, and others copied.  In most IDEs I've used it defaults to just a single *, and I think in this case its confusing clang-format and causing the weird breaks you are seeing.

**Comment by skliper** [comment]
The GSFC coding standard uses `**` in all the examples.  The development guide also.

**Comment by astrogeco** [comment]
That coding standard keeps haunting us!

**Comment by jphickey** [comment]
Is it really part of the coding standard or did they just happen to use ** in the examples?

There's a difference - in a standards document unless they use "shall"/"must" statements - then they are just examples, not requirements.  As a contributing author to some standards document, its frustrating when readers misinterpret examples as requirements.

Either way - its a case where we should really consider weighing compatibility with current industry standards vs. compatibility with some examples in a document written many years ago that may not have a real reason behind it other than being what was frequently used in code in that organization at that time (which itself I assume was nothing more than the personal preference of some original authors).

Not that I have a strong preference - from my POV it is simply a question of how much time will be spent manually fixing up the statements - neither Eclipse nor VS Code nor clang-format seem to work well with `**` comment blocks, requiring manual intervention to fix them all the time, yet they all generally work OK with auto-formatting the `*` comment blocks and require no manual fixups.

We can certainly keep the `**` - but it will cost more in terms of time and effort, vs. using the industry standard, that's all.

**Comment by astrogeco** [comment]
Random question, how does doxygen deal with the `**`?

**Comment by jphickey** [comment]
Doxygen seems generally OK with it, as long as the opening comment block starts with the correct flag (e.g. `/**`) then it seems to just ignore whatever chars are repeating on the left side of every line of the block.

**Comment by skliper** [comment]
> Is it really part of the coding standard or did they just happen to use `**` in the examples?

There is no requirement to use the `**` style.  I'm definitely in favor of the more tool-friendly comment style, `*`.

**Comment by jphickey** [comment]
It would be nice to also check for superfluous whitespace in the `CMakeLists.txt` files too - these currently have many lines with trailing whitespaces.  A simple sed/awk script applied to these files can strip this out.

**Comment by astrogeco** [comment]
> It would be nice to also check for superfluous whitespace in the `CMakeLists.txt` files too - these currently have many lines with trailing whitespaces. A simple sed/awk script applied to these files can strip this out.

Did a check in a bunch of files while I was at it.

**Pull request merged**